src/map_intf.ml

```diff
@@ -1455,8 +1455,7 @@ module type Map = sig
     -> f:('v option -> 'v)
     -> ('k, 'v, 'cmp) t
 
-  (** Returns the value bound to the given key, raising [Caml.Not_found] of [Not_found_s]
-      if none exists. *)
+  (** Returns [Some value] bound to the given key, or [None] if none exists. *)
```

Comment:
I'm not excited to drop the find_exn documentation, because we lose information about the precise exceptions thrown. But I fixed the comment separately. This should come out in in our public release shortly.